core/src/main/scala/eu/ostrzyciel/jelly/core/ConverterFactory.scala

```diff
@@ -33,67 +33,105 @@ trait ConverterFactory[
   TNode, TDatatype : ClassTag, TTriple, TQuad
 ]:
   import ConverterFactory.*
-  
+
   def decoderConverter: TDecConv
 
+  def decoderConverter(namespaceHandler: (name: String, iri: TNode) => Unit): TDecConv
+
   /**
    * Create a new [[TriplesDecoder]].
    * @param supportedOptions maximum supported options for the decoder. If not provided, this.defaultSupportedOptions
    *                         will be used. If you want to modify this (e.g., to specify an expected logical stream
    *                         type), you should always use this.defaultSupportedOptions.withXxx.
+   * @param namespaceHandler function to handle namespace declarations in the stream. The first argument is the
+   *                         namespace prefix (without a colon), the second is the IRI node.
    * @return decoder
    */
-  final def triplesDecoder(supportedOptions: Option[RdfStreamOptions] = None): 
+  final def triplesDecoder(
+    supportedOptions: Option[RdfStreamOptions] = None,
+    namespaceHandler: (String, TNode) => Unit = (_, _) => ()
```

Comment:
Consider factoring this out to a type alias. The default (no-op) could also be a static object.